infra/modules/azure_app_service_plan_autoscaler/locals.tf

```diff
@@ -0,0 +1,148 @@
+locals {
+
+  is_app_service  = var.app_service_name != null
+  is_function_app = var.function_app_name != null
+
+  base_name = local.is_app_service ? data.azurerm_linux_web_app.this[0].name : data.azurerm_linux_function_app.this[0].name
+  # base_name = regex("[^-]+-[^-]+-[^-]+-[^-]+-[^-]+", local.is_app_service ? data.azurerm_linux_web_app.this[0].name : data.azurerm_linux_function_app.this[0].name)
+  # number    = regex("[0-9]+$", local.is_app_service ? data.azurerm_linux_web_app.this[0].name : data.azurerm_linux_function_app.this[0].name)
+
+  # autoscale_name      = "${local.base_name}-as-${local.number}"
+  autoscale_name      = replace(replace(replace(local.base_name, "fn", "as"), "func", "as"), "app", "as")
```

Comment:
I'd keep this is let us replace the current autoscaler that are cut and pasted everywhere.

Reply:
The procedure to remove the existing autoscaler from the state and import it into the new one is long and useless, and often it tries to recreate it even if the target resource id stays the same for some reason (I did it several times during TF refactoring tasks)

In my opinion, the quickest way to do it is to replace the existing definition with this module and apply it ignoring the resource recreation/update